helm/oauth2-proxy/templates/deployment.yaml

```diff
@@ -70,6 +70,23 @@ spec:
           - --google-service-account-json=/google/service-account.json
         {{- end }}
         {{- end }}
+        {{- if eq (default "cookie" .Values.sessionStorage.type) "redis" }}
+          - --session-store-type=redis
+        {{- if .Values.sessionStorage.redis.password }}
+          - --redis-password={{ .Values.sessionStorage.redis.password }}
+        {{- end }}
+        {{- if eq (default "" .Values.sessionStorage.redis.clientType) "standalone" }}
```

Comment:
Why are the defaults from `values.yaml` being overwritten here?

Reply:
I think you are referring to line 78, right?
This is what we have used in our charts to protect against an incomplete end-user provided values file. 

For example, if the end user is removing the redis section, and is sending smtg like:
```yaml
sessionStorage:
  type: "redis"
  redis:
```
trying to "keeping it clean" then the installation will fail with something similar to `nil pointer evaluating interface {}.clientType` because it overrides the default values and can't work though the logic. 

By doing that `if eq (default "" .Values.sessionStorage.redis.clientType) "standalone"`, it will evaluate to false, which is correct since the provided value is not standalone.